src/test/java/org/sagebionetworks/bridge/dynamodb/DynamoSurveyDaoTest.java

```diff
@@ -627,8 +630,8 @@ public void getAllSurveysMostRecentVersionExcludeDeleted() {
     
     @Test(expectedExceptions = IllegalStateException.class, expectedExceptionsMessageRegExp = 
             "Calculated the need to query by secondary index, but study identifier is not set")
-    public void secondaryIndexRequiresStudyId() {
-        dao.versionSurvey(new DynamoSurvey());
+    public void secondaryIndexRequiresStudyIdWithStudyIdentifier() {
+        dao.versionSurvey(TEST_STUDY, new DynamoSurvey());
```

Comment:
This test is a little bit contrived. In practice, versionSurvey() will only ever be used to get a survey by guid+createdOn or by studyId+identifier+createdOn. I would rewrite this test to use getAllSurveysMostRecentVersion().

I would also go into SurveyService and add a checkNotNull() for survey guid and createdOn for any API that calls getSurvey(), such as get, delete, update, version, and publish.